server/src/server.rs

```diff
@@ -243,8 +242,8 @@ impl ServerStreamCallbacks {
                 frames
             }
             _ => {
-                debug!("Unexpected message {:?} during data_callback", r);
-                -1
+                error!("Unexpected message {:?} during data_callback", r);
+                0
```

Comment:
cubeb_wasapi doesn't handle errors from the user-supplied data_callback, it simply asserts.  That's likely true for other backends too, since cubeb never had tests to verify the backend behaviour in data_callback error situations.

cubeb_pulse and cubeb_audiounit (and the Rust ports) do attempt to handle these errors.  But note that since the handling is untested in cubeb and avoided in Gecko (since we never hit the cubeb_wasapi issue before), these code paths are effectively untested.

I filed a [bug](https://github.com/kinetiknz/cubeb/issues/553) to fix this in cubeb_wasapi and any other backends that handle this incorrectly, but it's riskier than changing the behaviour here so I'd prefer to work around the problem here and address it properly in cubeb_wasapi (and by adding tests to cubeb) in the future.  This change also makes ServerStreamCallbacks::data_callback match the existing error behaviour in data_cb_c, so it's not totally inconsistent.

I'll add a comment to clarify the situation too.